defi/lending_platform/src/lending_platform.rs

```diff
@@ -0,0 +1,436 @@
+                    self.assets.insert(asset_address, vault);
+                }
+            };
+
+            // Update User Deposit Balance
+            user.increase_deposit_balance(asset_address, amount);
+            self.users.insert(user_auth.resource_address(), user);
+            info!(
+                "[LendingPlatform][USER:{}] Depositing {} of {}.",
+                user_address,
+                amount,
+                asset_address
+            );
+
+            info!(
+                "[LendingPlatform][POOL] New liquidity balance for {} is {}.",
+                asset_address,
+                self.assets.get(&asset_address).unwrap().amount()
+            );
+            user_auth
+        }
+
+        pub fn withdrawal_asset(
+            &self,
+            user_auth: BucketRef,
+            asset_address: Address,
+            amount: Decimal
+        ) -> (BucketRef, Bucket) {
+            let mut user = self.users.get(&user_auth.resource_address()).unwrap();
+            let user_address = user.user_address;
+
+            let removed_asset = match self.assets.get(&asset_address) {
+                Some(x) =>  {
+                    assert! (
+                        x.amount() >= amount,
+                        "[LendingPlatform][POOL] Pool only has {} of asset {} but {} was requested",
+                        x.amount(),
+                        asset_address,
+                        amount
+                        );
+                    x.take(amount)
+                },
+                None => {
+                     panic!(
+                        "[LendingPlatform][POOL] Cannot decrease balance of asset {}. \
+                        No liquidity exists yet for it.",
+                        asset_address
+                    )
+                }
+            };
+
+            // Update User Deposit Balance
+            user.decrease_deposit_balance(asset_address, amount);
+            self.users.insert(user_auth.resource_address(), user);
+            info!(
+                "[LendingPlatform][USER:{}] Removing {} of {}.",
+                user_address,
+                amount,
+                asset_address
+            );
+
+            info!(
+                "[LendingPlatform][POOL] New liquidity balance for {} is {}.",
+                asset_address,
+                self.assets.get(&asset_address).unwrap().amount()
+            );
+
+            (user_auth, removed_asset)
+        }
+
+        pub fn borrow_asset(
+            &self,
+            user_auth: BucketRef,
+            asset_address: Address,
+            amount: Decimal
+        ) -> (BucketRef, Bucket) {
+            let mut user = self.users.get(&user_auth.resource_address()).unwrap();
+            let user_address = user.user_address;
+
+            let borrowed_asset = match self.assets.get(&asset_address) {
+                Some(x) =>  {
+                    assert! (
+                        x.amount() >= amount,
```

Comment:
Should you take into account the `asset_ltv_ratios` here to make sure the user is over-collateralizing their loan ?